cpp/src/io/parquet/experimental/dictionary_page_filter.cu

```diff
@@ -46,11 +48,1538 @@ using parquet::detail::ColumnChunkDesc;
+
+    // Offset into the running sum of values to be decoded from all column chunks of this column
+    auto const value_offset = value_offsets[row_group_idx];
+
+    auto buffer_offset      = int32_t{0};
+    auto num_values_decoded = key_type{0};
+
+    // Decode values from the current dictionary page
+    while (buffer_offset < page_data_size) {
+      // Check if we have a stream overrun
+      if (num_values_decoded > page.num_input_values) {
+        set_error(error, decode_error::DATA_STREAM_OVERRUN);
+        break;
+      }
+
+      // Decode cudf::string_view value
+      decoded_data[value_offset + num_values_decoded] =
+        decode_string_value(page_data, page_data_size, buffer_offset, error);
+
+      // Break if an error has been set within `decode_string_value`
+      if (is_error_set(error)) { break; }
+
+      // Insert the key (decoded value's global index) into the cuco hash set
+      set_insert_ref.insert(static_cast<key_type>(value_offset + num_values_decoded));
+
+      // Update the number of values decoded
+      num_values_decoded++;
+    }
+  }
+}
+
+/**
+ * @brief Decode fixed width column chunk dictionaries and build `cuco::static_set`s from them,
+ * one hash set per dictionary
+ *
+ * @tparam Supported underlying (fixed width) data type of the cudf column
+ * @param pages Column chunk dictionary page headers
+ * @param chunks Column chunk descriptors
+ * @param decoded_data Span of storage for decoded values from all dictionaries
+ * @param set_storage Pointer to the start of the bulk cuco hash set slots
+ * @param set_offsets Pointer to offsets into the bulk set storage for each dictionary
+ * @param value_offsets Pointer to offsets into running sum of values in each dictionary
+ * @param physical_type Parquet physical type of the column
+ * @param num_dictionary_columns Total number of columns with dictionaries
+ * @param dictionary_col_idx Index of the current dictionary column
+ * @param error Pointer to the kernel error code
+ */
+template <SupportedFixedWidthType T>
+CUDF_KERNEL void __launch_bounds__(DECODE_BLOCK_SIZE)
+  build_fixed_width_dictionaries(PageInfo const* pages,
+                                 ColumnChunkDesc const* chunks,
+                                 cudf::device_span<T> decoded_data,
+                                 slot_type* set_storage,
+                                 cudf::size_type const* set_offsets,
+                                 cudf::size_type const* value_offsets,
+                                 parquet::Type physical_type,
+                                 cudf::size_type num_dictionary_columns,
+                                 cudf::size_type dictionary_col_idx,
+                                 kernel_error::pointer error)
+{
+  // Each thread block (cg) decodes values from one dictionary page and inserts into the
+  // corresponding cuco hash set
+  auto const group = cg::this_thread_block();
```

Comment:
I'm curious why we're using cg here but not for strings. Seems like the cg approach is more flexible, especially considering the variable-ness of strings.